core/kernel.go

```diff
@@ -202,6 +205,44 @@ func (kern *Kernel) LoadDump(genesisDoc *genesis.GenesisDoc, restoreFile string,
 	return nil
 }
 
+// DialPeersFromGenesis allows us to autoconnect local validators on start
+func (kern *Kernel) DialPeersFromGenesis(nodes []string) {
+	pb := make(map[string]string, len(nodes))
+	for _, entry := range nodes {
+		target := strings.Split(entry, "=")
+		if len(target) != 2 {
+			kern.Logger.InfoMsg(fmt.Sprintf("Entry '%s' is not valid", entry))
+		}
+		pb[target[0]] = target[1]
+	}
+
+	validators := kern.Blockchain.GenesisDoc().Validators
+	for _, val := range validators {
+		if _, ok := pb[val.Name]; !ok {
+			continue
+		}
+		host, port, err := net.SplitHostPort(pb[val.Name])
+		if err != nil {
+			kern.Logger.InfoMsg(fmt.Sprintf("Cannot parse host '%s'", pb[val.Name]))
+		}
+		portNum, err := strconv.ParseUint(port, 10, 16)
+		if err != nil {
+			kern.Logger.InfoMsg(fmt.Sprintf("Cannot parse port '%s'", port))
+		}
+		nodeAddress := strings.ToLower(val.NodeAddress.String())
```

Comment:
`NodeAddress` is not always or by default available. So at the very least we need to fail early if we do not have access to the nodeaddress.